docs/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,258 @@
+
+## Branching strategy
+The project follows Red Hat Openshift Virtualization versions lifecycle.
+If needed, once your PR is merged to `main`, cherry-pick your PR to the relevant branch(es).
+
+
+## Python
+- Reduce duplicate code, before writing new function search for it, probably someone already wrote it or one that should serve your needs.
+  - The project uses external packages that may already have a functionality that does what you need.
+- When using a variable more than once, save it and reuse.
+- Keep functions and fixtures close to where they're used, if needed to move them later for more modules to use them.
+- Call functions using argument names to make it clear what is being passed and easier refactoring.
+- Imports: Always use absolute paths
+- Imports: when possible, avoid importing a module but rather import specific functions
+- Do not import from `conftest.py` files. These files must contain fixtures only and not utility functions, constants etc.
+- Avoid using nested functions.
+- Flexible code is good, but:
+  - Should not come at the expense of readability; remember that someone else will need to look/use/maintain the code.
+  - Do not prepare code for the future just because it may be useful.
+  - Every function, variable, fixture, etc. written in the code - must be used, or else removed.
+- Log enough to make you debug and understand the flow easy, but do not spam the log with unuseful info.
+Error logs should be detailed with what failed, status and so on.
+
+## Directory structure
+- Each feature should have its own subdirectory under the relevant component's subdirectory.
+- If needed, split feature tests into multiple files.
+- If needed, split into multiple subdirectories.
+- Each test should have its own file where the actual tests are written.
+- If helper utils are needed, they should be placed in the test's subdirectory.
+- If specific fixtures are needed, they should be placed in a `conftest.py` file under the test's subdirectory.
+
+
+## Interacting with Kubernetes/OpenShift APIs
+The project utilizes [openshift-python-wrapper](https://github.com/RedHatQE/openshift-python-wrapper).
+Please refer to the [documentation](https://github.com/RedHatQE/openshift-python-wrapper/blob/main/README.md)
+and the [examples](https://github.com/RedHatQE/openshift-python-wrapper/tree/main/examples) for more information.
+
+
+## conftest
+- Top level [conftest.py](../conftest.py) contains pytest native fixtures.
+- General tests [conftest.py](../tests/conftest.py) contains fixtures that are used in multiple tests by multiple teams.
+- If needed, create new `conftest.py` files in the relevant directories.
+
+
+## Fixtures
+- Ordering: Always call pytest native fixtures first, then session-scoped fixtures and then any other fixtures.
+- Fixtures should handle setup (and the teardown, if needed) needed for the test(s), including the creation of resources, for example.
+- Fixtures should do one thing only.
```

Comment:
What a "thing" represents is controversial and not well defined.
The reasoning this rule is set here is IMO more important than the examples, perhaps you can place them before.

I admit that I am not in agreement on this, but as a base rule I can live with the limitations.
Per my observation on the project, the fixtures are overused and in many cases missused with a lot of assumptions. (e.g. execution order per definition in a test).